pkgs/applications/editors/emacs/make-emacs.nix

```diff
@@ -88,7 +88,7 @@
 , withWebP ? lib.versionAtLeast version "29"
 , withX ? !(stdenv.isDarwin || noGui || withPgtk)
 , withXinput2 ? withX && lib.versionAtLeast version "29"
-, withXwidgets ? !stdenv.isDarwin && !noGui && (withGTK3 || withPgtk)
+, withXwidgets ? !stdenv.hostPlatform.isDarwin && !noGui && (withGTK3 || withPgtk) && (lib.versionOlder version "30") # XXX: upstream bug 66068 precludes newer versions of webkit2gtk (https://lists.gnu.org/archive/html/bug-gnu-emacs/2024-09/msg00695.html)
```

Comment:
`hostPlatform` seem not necessary

```console
> nix repl github:nixos/nixpkgs/nixos-24.05                                                                                                                             2:19Nix 2.24.8
Type :? for help.
Loading installable 'github:nixos/nixpkgs/nixos-24.05#'...
Added 5 variables.
nix-repl> legacyPackages.x86_64-linux.stdenv.isDarwin              
false

nix-repl> legacyPackages.x86_64-linux.stdenv.hostPlatform.isDarwin
false
```

Reply:
I did `hostPlatform` since the commit on `master` already had this as `hostPlatform` and I wanted the two branches to be consistent. Is it OK to have the branches diverge in this way? If so, I'm fine with omitting `hostPlatform` here to keep the diff clean.

Reply:
Oh, now I see why you add `hostPlatform`.  I do not have a strong opinion on this.